### PR TITLE
Frontend feat 1288 payment page

### PR DESCRIPTION
**Pull request by Vignesh-772**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

Payment page for YS

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

**Review by kavya-shree-s — CHANGES_REQUESTED** [review]

File: Frontend/ui-driver/src/Flow.purs [hunk 1]
```diff
@@ -1719,8 +1877,100 @@ homeScreenFlow = do
...
+setPaymentStatus paymentStatus (PayPayload payload) = do
+    case paymentStatus of
+      Success -> do
+                setValueToLocalStore SHOW_PAYMENT_MODAL "false"
+                modifyScreenState $ AcknowledgementScreenType (\a -> a { data {
+                  title = Just ( case getValueToLocalStore LANGUAGE_KEY of
+                        "EN_US" -> "Payment of ₹" <> payload.amount <>" Successful!"
+                        "HI_IN" -> "₹" <> payload.amount <> " का भुगतान सफल!"
+                        "BN_IN" -> "₹"<> payload.amount <> " পেমেন্ট সফল!"
+                        _       -> "Payment of ₹" <> payload.amount <>" Successful!"
```
Comment (Vignesh-772) [inline Frontend/ui-driver/src/Flow.purs:1929]: use currency from config

File: Frontend/output/Language.Types/externs.cbor
[the remark this answers is not in the document]
Comment (Rohit4997) [reply Frontend/output/Language.Types/externs.cbor:1]: @kavya-shree-s  it was already there, I deleted it

File: Frontend/ui-driver/index.js [hunk 2]
```diff
@@ -237,6 +231,25 @@ window["onEvent'"] = function (event, args) {
...
+  if ((JSON.parse(jsonPayload)).event == "initiate_result"){
+    callInitiateResult();
+    
+  }
+}
+
+function callInitiateResult () {
+  let payload = {
+    event: "initiate_result"
+    , service: "in.juspay.becknui"
```
[the remark this answers is not in the document]
Comment (Rohit4997) [reply Frontend/ui-driver/index.js:245]: @Vignesh-772  can you confirm?

[the remark this answers is not in the document]
Comment (Rohit4997) [reply Frontend/ui-driver/index.js:245]: @kavya-shree-s  we are not using it, It was already there we just created a function

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java [hunk 3]
```diff
@@ -410,7 +410,8 @@ private void initApp() {
             payload.put("clientId", getResources().getString(R.string.client_id));
             payload.put("action", "initiate");
             payload.put("service", getService());
-            payload.put(PaymentConstants.ENV, "master");
+            payload.put("merchantId", "yatrisathi");
```
Comment (kavya-shree-s) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java:413]: @Rohit4997 Why "merchantId" is hardcoded

File: Frontend/ui-driver/src/Components/PaymentHistoryModel/View.purs [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+          , width WRAP_CONTENT
+          , text $ getString DOWNLOAD_STATEMENT
+          , color Color.blue900
+          , margin $ MarginRight 12
+          ]
+        <> FontStyle.captions LanguageStyle
+    , imageView
+        [ height $ V 16
+        , width $ V 16
+        , imageWithFallback "ny_ic_download_button,https://assets.juspay.in/nammayatri/images/driver/ny_ic_download_button.png"
```
Comment (kavya-shree-s) [inline Frontend/ui-driver/src/Components/PaymentHistoryModel/View.purs:77]: @Rohit4997 Check this as well

File: Frontend/ui-driver/src/Flow.purs [hunk 5]
```diff
@@ -1715,8 +1873,101 @@ homeScreenFlow = do
...
+                setValueToLocalStore SHOW_PAYMENT_MODAL "false"
+                let time2PmTo10Am = (withinTimeRange "14:00:00" "10:00:00" (convertUTCtoISC(getCurrentUTC "") "HH:mm:ss"))
+                modifyScreenState $ AcknowledgementScreenType (\a -> a { data { title = Just (getString PAYMENT_PENDING), description = Just (getString PAYMENT_PENDING_DESC), primaryButtonText = Just  (getString GO_TO_HOME) , illustrationAsset = "ny_ic_payment_pending,"}, props {illustrationType = ST.Image, paymentStatus = paymentStatus}})
+                modifyScreenState $ HomeScreenStateType (\homeScreenState -> homeScreenState { data {paymentState {
+                  paymentStatus = paymentStatus,
+                  paymentStatusBanner = not (paymentStatus == Success),
+                  makePaymentModal = false,
+                  bannerBG = if time2PmTo10Am then "#FFECED" else "#FFFAEE",
+                  bannerTitle = getString if time2PmTo10Am then YOUR_PREVIOUS_PAYMENT_IS_PENDING else WE_WILL_NOTIFY_WHEN_PAYMENT_SUCCESS,
+                  bannerTitleColor = if time2PmTo10Am then "#BF4A4E" else "#FCB700",
```
Comment (kavya-shree-s) [inline Frontend/ui-driver/src/Flow.purs:1955]: @Rohit4997 Why code colour is used directly ?

File: Frontend/ui-driver/src/Flow.purs [hunk 6]
```diff
@@ -1715,8 +1873,101 @@ homeScreenFlow = do
...
+                  banneActionText = getString if time2PmTo10Am then CONTACT_SUPPORT else CONTINUE_TAKING_RIDES,
+                  bannerImage = if time2PmTo10Am then "ny_ic_payment_failed_banner," else "ny_ic_payment_pending_banner,",
+                  blockedDueToPayment = time2PmTo10Am,
+                  actionTextColor = if time2PmTo10Am then "#BF4A4E" else "#FCB700"
```
Comment (kavya-shree-s) [inline Frontend/ui-driver/src/Flow.purs:1959]: Here as well

File: Frontend/ui-driver/src/Flow.purs [hunk 7]
```diff
@@ -1715,8 +1873,101 @@ homeScreenFlow = do
...
+                  }}})
+
+
+      Failed -> modifyScreenState $ AcknowledgementScreenType (\a -> a { data { title = Just (getString PAYMENT_FAILED), description = Just (getString PAYMENT_FAILED_DESC), primaryButtonText = Just  "Retry Payment" , illustrationAsset = "ny_failed,"}, props {illustrationType = ST.Image, paymentStatus = paymentStatus}})
+
+      Pending -> do
+                setValueToLocalStore PAYMENT_STATUS_POOLING "true"
+                setValueToLocalStore SHOW_PAYMENT_MODAL "false"
+                let time2PmTo10Am = (withinTimeRange "14:00:00" "10:00:00" (convertUTCtoISC(getCurrentUTC "") "HH:mm:ss"))
+                modifyScreenState $ AcknowledgementScreenType (\a -> a { data { title = Just (getString PAYMENT_PENDING), description = Just (getString PAYMENT_PENDING_DESC), primaryButtonText = Just  (getString GO_TO_HOME) , illustrationAsset = "ny_ic_payment_pending,"}, props {illustrationType = ST.Image, paymentStatus = paymentStatus}})
```
Comment (kavya-shree-s) [inline Frontend/ui-driver/src/Flow.purs:1948]: Use proper name other than "a"

File: Frontend/ui-driver/src/Strings/KN.js [hunk 8]
```diff
@@ -564,4 +564,34 @@ const kannadaStrings = {
   MISSED_OPPORTUNITY : "ತಪ್ಪಿದ ಅವಕಾಶಗಳು",
   HOW_OLD_IS_YOUR_VEHICLE : "ನಿಮ್ಮ ವಾಹನ ಎಷ್ಟು ಹಳೆಯದು (ವರ್ಷಗಳಲ್ಲಿ)?",
   NEW_ : "ಹೊಸದು",
+  TODAY : "ಇಂದು",
+  OKAY : "ಸರಿ ಸರಿ",
+  NO_PAYMENT_HISTORY_AVAILABLE : "ಯಾವುದೇ ಪಾವತಿ ಇತಿಹಾಸ ಲಭ್ಯವಿಲ್ಲ",
+  YOU_DONT_HAVEANY_PAYMENTS : "ನೀವು ಇನ್ನೂ ಯಾವುದೇ ಪಾವತಿ ಮಾಡಿಲ್ಲ",
+  ENTER_AADHAAR_NUMBER : "ಆಧಾರ್ ಸಂಖ್ಯೆ/ಯುಐಡಿ ನಮೂದಿಸಿ",
+  ENTER_AADHAAR_OTP : "ಆಧಾರ್ OTP ನಮೂದಿಸಿ",
+  AADHAAR_LINKING_REQUIRED : "আধার লিঙ্ক করা আবশ্যক",
```
Comment (kavya-shree-s) [inline Frontend/ui-driver/src/Strings/KN.js:573]: Change language

File: Frontend/ui-driver/src/Strings/HI.js [hunk 9]
```diff
@@ -567,5 +567,59 @@ const hindiStrings = {
...
+  PAYMENT_PENDING_DESC : "चिंता न करें, आप अभी भी दिन भर सवारी लेना जारी रख सकते हैं!",
+  PAYMENT_FAILED_DESC : "आप भुगतान का पुनः प्रयास कर सकते हैं, या अपने निकटतम यात्री साथी बूथ पर भुगतान कर सकते हैं",
+  WE_WILL_NOTIFY_WHEN_PAYMENT_SUCCESS : "आपका भुगतान सफल होने पर हम सूचित करेंगे",
+  CONTINUE_TAKING_RIDES : "सवारी लेना जारी रखें",
+  YOUR_PREVIOUS_PAYMENT_IS_PENDING : "आपका पिछला भुगतान अभी भी लंबित है",
+  GOVERMENT_CHARGES : "सरकारी शुल्क",
+  OKAY : "ठीक है",
+  NO_PAYMENT_HISTORY_AVAILABLE : "कोई भुगतान इतिहास उपलब्ध नहीं है",
+  YOU_DONT_HAVEANY_PAYMENTS : "आपने अभी तक कोई भुगतान नहीं किया है",
+  ENTER_AADHAAR_NUMBER : "Enter Aadhaar Number/UID",
```
Comment (kavya-shree-s) [inline Frontend/ui-driver/src/Strings/HI.js:599]: Add translation

File: Frontend/ui-driver/src/Strings/KN.js (hunk 8 above)
Comment (kavya-shree-s) [reply Frontend/ui-driver/src/Strings/KN.js:573]: @Rohit4997

**Review by kavya-shree-s — APPROVED** [review]

**Pull request merged**